lib/ClangImporter/ClangImporter.cpp

```diff
@@ -533,6 +533,12 @@ getNormalInvocationArguments(std::vector<std::string> &invocationArgStrs,
       });
     }
   } else {
+    invocationArgStrs.insert(invocationArgStrs.end(), {
```

Comment:
This seems really random. @jrose-apple your thoughts?

Reply:
I would be ok with wrapping this in a if tripple.isFuchsia() here but it seemed benign on glibc libs. glibc and musl (and consequently our libc which is heavily forked from musl) hides or defines things in different locations depending on if BSD_SOURCE, GNU_SOURCE, etc are set because of features.h. If nothing is set then some pretty common functions go missing in musl libcs that are used in some of the platform folder code for the glibc module. Without some x_SOURCE define, glibc behavior is not exactly sync'd up with musl (or even bionic on android). Setting GNU_SOURCE seems to gets glibc and musl similar enough so that the end up defining types and declaring libc functions in the same places in the same named header files which makes the clang importer happy and things consistent with the glibc module.

There was thread about making this change on the list in 2016 by someone else and that someone would ok it if they sent a PR but nothing came of it.

Reply:
Consequently this change also gets arch-linux closer to working with swift since they use musl instead of glibc.

Reply:
I assume this is the previous thread:
https://lists.swift.org/pipermail/swift-dev/Week-of-Mon-20151221/000541.html

Pierre and Jordan were fine with it for Linux then.

Reply:
https://bugs.swift.org/browse/SR-328

Reply:
Yeah, I don't love it, but it seems reasonable. 👍 from me.

Reply:
One issue I still need to solve is that Swift bakes in absolute paths for the glibc module map, which for cross compiling where the sysroot of the platform you are targeting can be at a different location on each developer's system, is big a non-starter. It basically means I can't make any binary swift cross-compile capable toolchains that are reusable. Having a "-sdk"/sysroot relative module map path lookup would be a good work around but it would require some deeper changes to the clang importer and module code in clang to support that concept. 

There was a discussion about this last year on the list and an abandoned PR that was approaching fixing this but the only use case that was fixed was for for canadian cross-compiling any not for direct cross compiling. 

Right now I've just made a platform libc module map in our Fuchsia sysroot that lives relative to all our headers which are internal swift code is using, but the glibc module doesn't work for anyone that I share my swift toolchain builds with.

Reply:
Let's tackle that later, in a separate PR.